tests/components/websocket_api/test_commands.py

```diff
@@ -666,6 +668,190 @@ async def test_subscribe_unsubscribe_events_state_changed(
     assert msg["event"]["data"]["entity_id"] == "light.permitted"
 
 
+async def test_subscribe_unsubscribe_entities(hass, websocket_client, hass_admin_user):
+    """Test subscribe/unsubscribe entities."""
+
+    STATE_KEY_LONG_NAMES = {v: k for k, v in STATE_KEY_SHORT_NAMES.items()}
+
+    def _apply_changes(state_dict: dict, change_dict: dict) -> None:
+        """Apply a diff set to a dict.
+
+        Port of the client side merging
+        """
+        additions = change_dict.get("+", {})
+        if "lc" in additions:
+            additions["lu"] = additions["lc"]
+        sub_dicts = [
+            key for key, change in additions.items() if isinstance(change, dict)
+        ]
+        for key in sub_dicts:
+            state_dict[STATE_KEY_LONG_NAMES[key]].update(additions.pop(key))
+        for k, v in additions.items():
+            state_dict[STATE_KEY_LONG_NAMES[k]] = v
+        for key, items in change_dict.get("-", {}).items():
+            for item in items:
+                del state_dict[STATE_KEY_LONG_NAMES[key]][item]
+
+    hass.states.async_set("light.permitted", "off", {"color": "red"})
+    original_state = hass.states.get("light.permitted")
+    assert isinstance(original_state, State)
+    state_dict = {
+        "attributes": dict(original_state.attributes),
+        "context": dict(original_state.context.as_dict()),
+        "entity_id": original_state.entity_id,
+        "last_changed": original_state.last_changed.isoformat(),
+        "last_updated": original_state.last_updated.isoformat(),
+        "state": original_state.state,
+    }
+    hass_admin_user.groups = []
+    hass_admin_user.mock_policy({"entities": {"entity_ids": {"light.permitted": True}}})
+
+    await websocket_client.send_json({"id": 7, "type": "subscribe_entities"})
+
+    msg = await websocket_client.receive_json()
+    assert msg["id"] == 7
+    assert msg["type"] == const.TYPE_RESULT
+    assert msg["success"]
+
+    msg = await websocket_client.receive_json()
+    assert msg["id"] == 7
+    assert msg["type"] == "event"
+    assert msg["event"] == {
+        "add": {
+            "light.permitted": {
+                "a": {"color": "red"},
+                "c": {"id": ANY, "parent_id": None, "user_id": None},
```

Comment:
If we want we can compress this further by allowing `"c": "some-id"` and interpreting `"some-id"` as `{"id": "some-id", "parent_id": None, "user_id": None}`

Reply:
implemented